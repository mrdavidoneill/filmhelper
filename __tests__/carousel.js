import "@testing-library/jest-dom";

import React from "react";
import { render, waitFor } from "@testing-library/react";
import Carousel from "@/components/carousel";

describe("Carousel", () => {
  it("should match the snapshot", async () => {
    const images = ["image1.jpg", "image2.jpg"];
    const setCurrentIndex = jest.fn();
    const initialIndex = 0;

    const { container, getByTestId } = render(
      <Carousel
        images={images}
        setCurrentIndex={setCurrentIndex}
        initialIndex={initialIndex}
      />
    );

    await waitFor(() => {
      expect(getByTestId("image1.jpg")).toBeInTheDocument();
      expect(getByTestId("image2.jpg")).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
