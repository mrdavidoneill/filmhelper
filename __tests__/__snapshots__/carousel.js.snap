// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match the snapshot 1`] = `
<div>
  <div
    class="w-full animate-fadeinslow"
  >
    <div
      class="bg-black cursor-pointer outline-white flickity-enabled is-draggable"
      tabindex="0"
    >
      <div
        class="flickity-viewport"
        style="height: 0px;"
      >
        <div
          class="flickity-slider"
          style="left: 0px; transform: translateX(NaN%);"
        >
          <div
            class="flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center is-selected"
            data-testid="image1.jpg"
            style="background-image: url(image1.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
          />
          <div
            aria-hidden="true"
            class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
            data-testid="image2.jpg"
            style="background-image: url(image2.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
